### ISSUE TITLE
Function "makeUserEditCountDataItem" cuases error on user subpages

### ISSUE THREAD

**Issue by smthmsn** [SemanticMediaWiki/SemanticExtraSpecialProperties#49]
Issue: 
If _USEREDITCNT is enabled, the error below is thrown while editing or creating a user subpage (/User:Foo/newSubPage)

Error Thrown:
Fatal error: Call to a member function getEditCount() on a non-object in ..\extensions  \SemanticExtraSpecialProperties\src\Annotator\ExtraPropertyAnnotator.php on line 360

Explanation: 
It looks like the current function assumes that the $user will exist, but in the case of a user subpage the $user variable returns false. 

Fix: 
I was able to resolve the issue by adding another check after isUserPage(), to check isSubpage().
```
private function makeUserEditCountDataItem() {
	
	if ( !$this->isUserPage() || $this->getWikiPage()->getTitle()->isSubpage() ) {
			return;
		}

		$user = $this->appFactory->newUserFromTitle( $this->getWikiPage()->getTitle() );

		$count = $user->getEditCount();
		
		if ( $count ) {

			return new DINumber( $count );
		}
	}
```
Current Version Info:
MediaWiki - 1.23.6
SESP - 1.3.0
SMW - 2.1

**Comment by kghbln** [comment]
@smthmsn Thank you for reporting this issue as well as providing a fix. 

@mwjames Do you think the fix is cool? Looks reasonable to me.

**Comment by mwjames** [comment]
>  Do you think the fix is cool? Looks reasonable to me.

It will work but instead of making this function depend on a subpage check, use the instance for validation (it is about the user not a subpage which of course could be used as scapegoat) therefore:

```php
		if ( !$this->isUserPage() ) {
			return;
		}

		$user = $this->appFactory->newUserFromTitle( $this->getWikiPage()->getTitle() );

		$count = $user instanceof User ? $user->getEditCount() : false;

		if ( $count ) {
			return new DINumber( $count );
		}
```

**Comment by smthmsn** [comment]
That works in my environment as well. Thanks for the quick response guys.

**Comment by kghbln** [comment]
@smthmsn Thanks for doing the field test!

@mwjames Thanks for working on the patch provided! I have done a pull request https://github.com/SemanticMediaWiki/SemanticExtraSpecialProperties/pull/50. What about doing a 1.3.1 release for this? Will add the fix to the RELEASE NOTES when doing it.

**Comment by kghbln** [comment]
Fix released with [version 1.3.1](https://github.com/SemanticMediaWiki/SemanticExtraSpecialProperties/blob/master/RELEASE-NOTES.md#131). Thanks to all of you!
